libraries/core_libs/consensus/include/transaction/gas_pricer.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <boost/circular_buffer.hpp>
+
+#include "final_chain/final_chain.hpp"
+
+namespace taraxa {
+/**
+ * @brief Basic gas price calculator. We calculate the gas price based on the lowest price in last number_of_blocks
+ * blocks. Then all of those values are sorted and currnet price is selected based on percentile
+ *
+ */
+class GasPricer {
+ public:
+  GasPricer(uint64_t percentile = 60, uint64_t number_of_blocks = 200, std::shared_ptr<DbStorage> db = {});
+  ~GasPricer();
+
+  /**
+   * @brief returns current gas price
+   *
+   * @return u256 gas price
+   */
+  u256 bid() const;
+
+  /**
+   * @brief updates gas price after each executed block
+   *
+   * @param trxs from latest block
+   */
+  void update(const Transactions& trxs);
+
+ private:
+  /**
+   * @brief initialize values on start from blocks in DB
+   *
+   * @param db
+   */
+  void init(std::shared_ptr<DbStorage> db);
+
+  const uint64_t percentile_;
+
+  mutable std::shared_mutex mutex_;
+  u256 latest_price_ = 1;
+  boost::circular_buffer<u256> price_list_;
```

Comment:
question is: do we want to use boost in general or do we want to minimize its usage ? I believe that except the dag we could get rid of boost pretty much everywhere if we wanted to, which would be cool

Reply:
? we use boost almost everywhere. Our logging is boost, threads (std does not have upgradable lock), asio, networking ...

Reply:
ok nevermind